nixos/modules/services/backup/borgbackup.nix

```diff
@@ -99,7 +99,17 @@ let
         BORG_REPO = cfg.repo;
         inherit (cfg) extraArgs extraInitArgs extraCreateArgs extraPruneArgs;
       } // (mkPassEnv cfg) // cfg.environment;
-      inherit (cfg) startAt;
+    };
+
+  mkBackupTimers = name: cfg:
+    nameValuePair "borgbackup-job-${name}" {
+      description = "BorgBackup job ${name} timer";
+      timerConfig = {
+        Persistent = cfg.persistentTimer;
+        OnCalendar = cfg.startAt;
```

Comment:
sorry to be at this again, but having tested a few things locally this no longer looks correct. systemd refuses timer units with no triggers (which could happen if `startAt == []`, which is totally valid). we should probably generate a timer unit only for backup jobs that have `startAt != []`. :slightly_frowning_face:

Reply:
having `startAt == []` means the service is available but has to be triggered manually?

Reply:
either that, or you can use the `borg-job-${name}` wrapper that is provided for each configured job

Reply:
sorry, but this is my first nixos-option i am doing. I might need some more guidance on how to do that.

could we use something like `(filterAttrs (cfg: cfg.startAt != []) cfg.services)` or what are more fitting functions?

should the filtering be done at line 104? or in 722?

Reply:
you could filter in either place, but in 722 it makes for a definition that's a bit easier to understand. in 104 you'd have to wrap the value in a `mkIf (cfg.startAt != [])`, which looks really odd and like spooky action at a distance :) in 722 it'd just be changing `jobs` to `(filterAttrs (cfg: cfg.startAt != []) jobs)`, which seems preferrable for clarity of what happens.

Reply:
i added a little comment as well

Reply:
looks good :)